### PR TITLE
Add merkle sum circuit

### PR DESCRIPTION
**Pull request by EkamSinghPandher**
1. Added merkle sum tree circuit
2. Added tests for circuit
3. Added parser for json parsing
4. Added prover for proving merkle sum circuits

### CODE REVIEW

File: crates/zk-por-api/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+[features]
+default = []
+
+[dependencies]
+plonky2={workspace=true}
+plonky2_field={workspace=true}
+
+[build-dependencies]
+bindgen = { version = "0.68.1" }
```
Comment (cliff0412) [inline crates/zk-por-api/Cargo.toml:24]: this is not used i think

File: crates/zk-por-core/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+[dependencies]
+plonky2={workspace=true}
+plonky2_field={workspace=true}
+log = "0.4.17"
+serde={workspace=true}
+serde_json={workspace=true}
+rayon={workspace=true}
+
+[build-dependencies]
+bindgen = { version = "0.68.1" }
```
Comment (cliff0412) [inline crates/zk-por-core/Cargo.toml:28]: here also

**Comment by cliff0412** [comment]
can squash all commits and have one commit message like ' feature:add-merkle-sum-tree circuit and test'

File: crates/zk-por-api/Cargo.toml (hunk 1 above)
Comment (EkamSinghPandher) [reply crates/zk-por-api/Cargo.toml:24]: Let me check

**Pull request merged**